src/physics/base/PhysicsParams.hh

```diff
@@ -0,0 +1,155 @@
+
+namespace celeritas
+{
+class MaterialParams;
+class ParticleParams;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Manage physics processes and models.
+ *
+ * The physics params takes a vector of processes and sets up the processes and
+ * models. It constructs data and mappings of data:
+ * - particle type and process to tabulated values of cross sections etc,
+ * - particle type to applicable processes
+ *
+ * During construction it constructs models and their corresponding list of
+ * \c ModelId values, as well as the tables of cross section data.
+ */
+class PhysicsParams
+{
+  public:
+    //!@{
+    //! Type aliases
+    using SPConstParticles   = std::shared_ptr<const ParticleParams>;
+    using SPConstMaterials   = std::shared_ptr<const MaterialParams>;
+    using SPConstProcess     = std::shared_ptr<const Process>;
+    using VecProcess         = std::vector<SPConstProcess>;
+    using SpanConstProcessId = Span<const ProcessId>;
+    //!@}
+
+    //! Global physics configuration options
+    struct Options
+    {
+        real_type max_step_over_range; //!< alpha_r, limit step range at high E
+        real_type min_step;            //!< rho_R, minimum range at low E
+    };
+
+    //! Physics parameter construction arguments
+    struct Input
+    {
+        SPConstParticles particles;
+        SPConstMaterials materials;
+        VecProcess       processes;
+
+        Options options;
+    };
+
+  public:
+    // Construct with processes and helper classes
+    explicit PhysicsParams(Input);
+
+    //// HOST ACCESSORS ////
+
+    //! Number of models
+    size_type num_models() const { return models_.size(); }
+
+    //! Number of processes
+    size_type num_processes() const { return processes_.size(); }
+
+    //! Number of particle types
+    size_type num_particles() const { return processes_by_particle_.size(); }
+
+    //! Maximum number of processes that apply to any one particle
+    size_type max_particle_processes() const { return max_pbp_; }
+
+    // Get a model
+    inline const Model& model(ModelId) const;
+
+    // Get a process
+    inline const Process& process(ProcessId) const;
+
+    // Get the processes that apply to a particlular particle
+    inline SpanConstProcessId processes(ParticleDefId) const;
+
+    //// DEVICE ACCESSORS ////
+
+    // Get managed data
+    PhysicsParamsPointers device_pointers() const;
+
+  private:
+    using SPConstModel = std::shared_ptr<const Model>;
+
+    //// HOST DATA ////
+
+    VecProcess                          processes_;
+    std::vector<std::vector<ProcessId>> processes_by_particle_;
+    std::vector<SPConstModel>           models_;
+    size_type                           max_pbp_ = 0;
+
+    //// DEVICE DATA ////
+
+    DeviceVector<UniformGridPointers> grid_inputs_;
+    DeviceVector<real_type>           xs_;
+    DeviceVector<real_type>           energy_loss_;
+    DeviceVector<real_type>           range_;
```

Comment:
This also was from an earlier iteration where these guys were all in separate vectors. I think you're right and it makes sense to store them all contiguously, especially since they're all being referenced by `XsGridPointers`.